### PR TITLE
Refactor RpcCall

### PR DESCRIPTION
**Pull request by gengliqi**
1. use local variable instead of `shared_ptr` for rpc request and response
2. replace original implementation of `RpcTypeTraits` with a more elegant way

### CODE REVIEW

File: include/pingcap/kv/RegionClient.h [hunk 1]
```diff
@@ -36,15 +36,15 @@ struct RegionClient
     {}
 
     // This method send a request to region, but is NOT Thread-Safe !!
-    template <typename T>
-    auto sendReqToRegion(Backoffer & bo,
-                         std::shared_ptr<T> req,
+    template <typename T, typename U, typename V>
```
Comment (breezewish) [inline include/pingcap/kv/RegionClient.h:39]: how about naming as typename REQ, RESP for better readibility?

**Review by breezewish — APPROVED** [review]
Mostly LGTM

File: include/pingcap/kv/RegionClient.h (hunk 1 above)
Comment (gengliqi) [reply include/pingcap/kv/RegionClient.h]: Done.

**Pull request merged**